### DISCUSSION TITLE
[General]: Using FormBuilder with Stepper

### DISCUSSION THREAD

**Discussion by devwebapps** [danvick/flutter_form_builder#1421] [Q&A]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Package/Plugin version

9.3.0

### What you'd like to happen

```Markdown
Hi.
I'm literally in despair. I want to use the FormBuilder with the Material Stepper Widget.
Is it possible that you could add an example in your documentation on how to use the FormBuilder with the stepper?

What do you have to do so that, for example, validation takes place in every step and no empty radio buttons are forwarded when you click on the next button?

How to deal with the Global Form Key? Do you create a separate FormBuilder widget with a new global form key for each step?

I hope I'm not the first/last person who would like to use your FormBuilder with the stepper, so an example would be helpful for others too.
```

**Comment by deandreamatias** [comment]
@devwebapps in example you have this examples https://github.com/flutter-form-builder-ecosystem/flutter_form_builder/blob/main/example/lib/sources/custom_fields.dart
Also, you can see the existed implementations, like https://github.com/flutter-form-builder-ecosystem/flutter_form_builder/blob/main/lib/src/fields/form_builder_slider.dart

**Reply by devwebapps** [reply]
I'm irritated. In none of the links can I find an example of how I can create a form divided into several steps using the FormBuilder and the Stepper Widget.

**Reply by deandreamatias** [reply]
Ok, I don't understand correctly in first time.
I think that you need a form with multiple steps/pages. E.g.: first page name, second page address and third page to confirm this infos.
The best way that I think, is create a separated forms on each page, with a formKey for each. Also, use a MVVM pattern to create a ViewModel that manage this formKeys and save data for use when finish.

Is more a solution about state management and architecture that a problem with FlutterFormBuilder
